### PR TITLE
ref task #130796 - Handle notification

### PR DESCRIPTION
**Pull request by NguyenTienVinh**
Hiển thị màn hình list request tương ứng khi kích vào notification (Notification nhận từ server thông qua Firebase Cloud Messaging)
https://dev.framgia.com/redmine/issues/130796
Effect : None

### CODE REVIEW

File: app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java [hunk 1]
```diff
@@ -353,6 +387,32 @@ public boolean onItemSelected(@NonNull MenuItem item) {
         return true;
     }
 
+    private void getPageNotification(String notificationRequestType) {
+        if (StringUtils.isBlank(notificationRequestType)) return;
```
Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: WARNING: 'if' construct must use '{}'s.

Comment (trainer-android-fr) [inline app/src/main/java/com/framgia/wsm/screen/main/MainViewModel.java]: WARNING: [IfStmtsMustUseBraces] Avoid using if statements without curly braces
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/braces.html#IfStmtsMustUseBraces

File: app/fabric.properties [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#Contains API Secret used to validate your application. Commit to internal source control; avoid making secret public.
+#Sun Jul 30 19:27:46 ICT 2017
+apiSecret=488599d73d56b6c48144d11adebcb6025e3bcae6086786347e5110d176a64499
```
Comment (daolq3012) [inline app/fabric.properties]: pls remove this file

File: app/src/main/java/com/framgia/wsm/data/source/remote/api/service/FireBaseMessageService.java [hunk 3]
```diff
@@ -25,31 +27,36 @@
...
+    private static final String LOCATION = "location";
 
     @Override
     public void onMessageReceived(RemoteMessage remoteMessage) {
         if (remoteMessage.getData().size() > 0) {
-            sendNotification(remoteMessage.getData().get(MESSAGE));
-        }
-        if (remoteMessage.getNotification() != null) {
-            sendNotification(remoteMessage.getNotification().getBody());
+            if (SHOW.equals(remoteMessage.getData().get(TYPE))) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/source/remote/api/service/FireBaseMessageService.java]: pls define a variable for ``remoteMessage.getData().get(TYPE)`` and use it in here!

File: app/src/main/java/com/framgia/wsm/data/source/remote/api/service/FireBaseMessageService.java [hunk 4]
```diff
@@ -25,31 +27,36 @@
...
+                String requestType = remoteMessage.getData().get(LOCATION);
+                sendNotification(remoteMessage.getData().get(MESSAGE), requestType);
+            } else {
+                sendNotification(remoteMessage.getData().get(MESSAGE), Constant.BLANK);
+            }
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/data/source/remote/api/service/FireBaseMessageService.java]: pls optimize this code to readable
like
```java
 String body = ...
String type = SHOW.equals(..)?remoteMessage.getData().get(LOCATION):Constan.BLANK;
sendNotification(body,type)
```

File: app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java [hunk 5]
```diff
@@ -58,6 +62,10 @@ public void run() {
         };
     }
 
+    public boolean isOpenNotification() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/main/MainActivity.java]: pls rename this method to ``isOpenAppByClickNotification``

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ListRequestFragment.java [hunk 6]
```diff
@@ -58,6 +58,11 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     public void onStart() {
         super.onStart();
         mViewModel.onStart();
+        MainActivity mainActivity = (MainActivity) getActivity();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ListRequestFragment.java]: pls change to
```java
if(getActivity() instanceOf MainActivity){
...
}

File: app/src/main/java/com/framgia/wsm/screen/listrequest/ListRequestFragment.java [hunk 7]
```diff
@@ -58,6 +58,11 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
...
+        boolean isOpenAppByClickNotification = mainActivity.isOpenAppByClickNotification();
+        if (isOpenAppByClickNotification) {
+            mViewModel.onReloadData(getArguments().getInt(Constant.EXTRA_REQUEST_TYPE));
+        }
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/wsm/screen/listrequest/ListRequestFragment.java]: move all line to onCreateView method of fragment
Pls check lifecycle of Fragment and Activity.

**Pull request merged**